src/mac.rs

```diff
@@ -627,38 +728,168 @@ impl Address {
     ///
     /// # Panics
     ///
-    /// Panics, if the buffer is less than 4 bytes long. If you believe that
+    /// Panics, if the buffer is less than 2 bytes long. If you believe that
     /// this behavior is inappropriate, please leave your feedback on the
     /// [issue tracker].
     ///
     /// # Example
     ///
     /// ``` rust
-    /// use ieee802154::mac::Address;
+    /// use ieee802154::mac::ShortAddress;
     ///
-    /// let address = Address { pan_id: 0x1234, short_addr: 0x5678 };
+    /// let address = ShortAddress(0x5678);
     ///
-    /// let mut bytes = [0u8; 4];
+    /// let mut bytes = [0u8; 2];
     /// address.encode(&mut bytes);
     ///
-    /// let expected_bytes = [0x34, 0x12, 0x78, 0x56];
+    /// let expected_bytes = [0x78, 0x56];
     /// assert_eq!(bytes[..expected_bytes.len()], expected_bytes[..]);
     /// ```
     ///
     /// [issue tracker]: https://github.com/braun-robotics/ieee-802.15.4/issues/9
     pub fn encode(&self, buf: &mut [u8]) -> usize {
-        let mut len = 0;
+        LittleEndian::write_u16(buf, self.0);
+        size_of_val(&self.0)
+    }
+}
 
-        LittleEndian::write_u16(&mut buf[len..], self.pan_id);
-        len += size_of_val(&self.pan_id);
+/// An address consisting of PAN ID and extended address
```

Comment:
Comment is wrong. The struct only contains the extended address, not the PAN ID.

Reply:
Yes, will fix.